src/Ng/CommandHelpers.cs

```
@@ -153,167 +245,150 @@ public static int GetInterval(IDictionary<string, string> arguments)
 
-            string storagePath = null;
-            arguments.TryGetValue("-compressedStoragePath", out storagePath);
+                string storageKeyValue;
+                TryGetArgument(arguments, CompressedStorageKeyValue, out storageKeyValue, required: true);
 
-            StorageCredentials credentials = new StorageCredentials(storageAccountName, storageKeyValue);
-            CloudStorageAccount account = new CloudStorageAccount(credentials, true);
-            return new AzureStorageFactory(account, storageContainer, storagePath, storageBaseAddress) { Verbose = verbose, CompressContent = true };
-        }
+                string storageContainer;
+                TryGetArgument(arguments, CompressedStorageContainer, out storageContainer, required: true);
 
-        public static StorageFactory CreateStorageFactory(IDictionary<string, string> arguments, bool verbose)
-        {
-            Uri storageBaseAddress = null;
-            string storageBaseAddressStr;
-            if (arguments.TryGetValue("-storageBaseAddress", out storageBaseAddressStr))
-            {
-                storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
+                string storagePath = null;
+                TryGetArgument(arguments, CompressedStoragePath, out storagePath);
 
-                storageBaseAddress = new Uri(storageBaseAddressStr);
+                StorageCredentials credentials = new StorageCredentials(storageAccountName, storageKeyValue);
+                CloudStorageAccount account = new CloudStorageAccount(credentials, true);
+                return new AzureStorageFactory(account, storageContainer, storagePath, storageBaseAddress) { Verbose = verbose, CompressContent = true };
             }
-
-            string storageType;
-            if (!arguments.TryGetValue("-storageType", out storageType))
+            catch (ArgumentException)
```

>ArgumentException [](start = 19, length = 17)

please check this, but I think that when a required argument is missing, you can just throw the exception and catch it higher in the code. From what I saw, the method that calls CreateCompressedStorageFactory, checks if the returned value is null and fails. It might as well catch an exception.